typing/btype.ml

```diff
@@ -57,14 +57,36 @@ module TransientTypePairs =
     let hash (t, t') = t.id + 93 * t'.id
  end)
 module TypePairs = struct
-  include TransientTypePairs
+  module H = TransientTypePairs
   open Transient_expr
-  let wrap_repr f (t1, t2) = f (repr t1, repr t2)
-  let wrap_type_expr f (tt1, tt2) = f (type_expr tt1, type_expr tt2)
-  let add hash = wrap_repr (add hash)
-  let find hash = wrap_repr (find hash)
-  let mem hash = wrap_repr (mem hash)
-  let iter f = TransientTypePairs.iter (wrap_type_expr f)
+
+  type t = {
+    mutable elems : (transient_expr * transient_expr) list;
+    set : unit H.t
+  }
```

Comment:
To an outside reader, it is not obvious why you are careful to also preserve this list, and why you reverse it in the 'iter' function below. Could you maybe add a comment that clarifies the intent?

Reply:
Sure, done.